nixos/modules/services/matrix/synapse.nix

```diff
@@ -151,13 +162,45 @@ in {
 
       package = mkOption {
         type = types.package;
-        default = pkgs.matrix-synapse;
-        defaultText = literalExpression "pkgs.matrix-synapse";
+        default = pkgs.matrix-synapse-unwrapped;
+        defaultText = literalExpression "pkgs.matrix-synapse-unwrapped";
         description = lib.mdDoc ''
           Overridable attribute of the matrix synapse server package to use.
         '';
       };
 
+      extras = mkOption {
+        type = types.listOf (types.enum (lib.attrNames cfg.package.optional-dependencies));
+        default = [
+          "postgres"
+          "systemd"
+          "url-preview"
+          "user-search"
+        ];
```

Comment:
Doesn't that mean that if you add `extras`, the defaults will be discarded?

Reply:
No, list options are additive. You'd need to `lib.mkForce` to overwrite the defaults.

Reply:
but
```nix
with import <nixpkgs/lib>;
(evalModules {
  modules = [
    {
      options.kek = mkOption {
        type = types.listOf types.str;
        default = [ "a" "b" ];
      };
    }
    {
      kek = [ "c" ];
    }
  ];
}).config.kek
```

evaluates to

`["c"]`

Reply:
Ah, so maybe they're `lib.mkDefault`, and adding some overwrites them, but writing from multiple places is additive again.

Reply:
`lib.mkOptionDefault` (lower prio than `lib.mkDefault`) to be precise.


Personally I'd prefer to list the defaults in `defaultText` and add them in the `config` section. That way it's actually additive and if you want something completely different, you can still use `mkForce`.